src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -22,6 +22,16 @@ def _resolve_aliases(target):
     else:
       yield declared
 
+    for export in _resolve_exports(declared):
+      yield export
+
+
+def _resolve_exports(target):
+  for export in getattr(target, 'exports', []):
+    for exp in _resolve_exports(export):
```

Comment:
Hm. After reading that thread, I think I changed my mind. I hadn't thought it through.

1. This is the case we're shooting for either way
```
A -d-> B
B -e-> C

classpath(A) => B, C
```
2. This is the transitive case
```
A -d-> B
B -e-> C
C -e-> D
# should either
#  only include direct exports
classpath(A) => B, C
#  or, include transitive exports
classpath(A) => B, C, D
```

* If we make exports transitive, `strict_deps` makes dependencies intransitive, but not exports. I think that's reasonable, in that strict_deps applies to dependencies not exports.
* For the classfile parsing case, eg if D provides an annotation, will we get a warning/error if it is not on the classpath along with C? I think we might get an error.
* For the inheritance case, making exports transitive means that if you inherit from C in A, that scalac's LUB check will be able to find D, which is an outcome we want.